mesa/space.py

```diff
@@ -62,65 +97,75 @@ def __init__(self, height, width, torus):
     def __getitem__(self, index):
         return self.grid[index]
 
-    def _get_x(self, x):
-        '''
-        Convert X coordinate, handling torus looping.
-        '''
-        if x >= 0 and x < self.width:
-            return x
-        if not self.torus:
-            raise Exception("Coordinate out of bounds.")
-        else:
-            return x % self.width
-
-    def _get_y(self, y):
-        '''
-        Convert Y coordinate, handling torus looping.
-        '''
-        if y >= 0 and y < self.height:
-            return y
-        if not self.torus:
-            raise Exception("Coordinate out of bounds.")
-        else:
-            return y % self.height
-
-    def get_neighborhood(self, x, y, moore, include_center=False, radius=1):
-        '''
-        Return a list of cells that are in the neighborhood of a certain point.
+    def __iter__(self):
+        # create an iterator that chains the
+        #  rows of grid together as if one list:
+        return itertools.chain(*self.grid)
+
+
+    def occupied_iter(self, occupied=True):
```

Comment:
@gcallah please add doc string & maybe we should consider updating the name on this also? Thoughts?